Makefile

```diff
@@ -6,95 +6,211 @@ ifneq ($(OS),Windows_NT)
 	OS := $(shell uname)
 endif
 
+.DEFAULT_GOAL := help
+
+.ONESHELL:
+
+.SHELLFLAGS = -ec
+
 .PHONY: help
 help: ## Show this help.
-	@printf "%-30s %s\n" "Target" "Description"
-	@printf "%-30s %s\n" "------" "-----------"
-	@fgrep " ## " $(MAKEFILE_LIST) | fgrep -v grep | awk -F ': .*## ' '{$$1 = sprintf("%-30s", $$1)} 1'
+	@printf "%-41s %s\n" "Target" "Description"
+	@printf "%-41s %s\n" "------" "-----------"
+	@fgrep " ##" $(MAKEFILE_LIST) | fgrep -v grep | sed 's/:[^#]*/ /' | awk -F '[: ]*' \
+	'{
+		if ($$2 == "##")
+		{
+			$$1=sprintf("%-40s", $$1);
+			$$2="";
+			print $$0;
+		}
+		else
+		{
+			$$1=sprintf(" └%-38s", $$1);
+			$$2="";
+			print $$0;
+		}
+	}'
+
+---------------- : ## ----------------
 
 .PHONY: setup
-setup: ## Set up a development environment
-ifeq ($(OS),Linux)
-	sudo snap install codespell ruff shellcheck
-	sudo snap install --classic astral-uv
-	sudo apt-get --yes install libxml2-dev libxslt-dev
-else ifeq ($(OS),Windows_NT)
-	pipx install uv
-	choco install shellcheck
-else ifeq ($(OS),Darwin)
-	bash -c 'curl -LsSf https://astral.sh/uv/install.sh | sh'
-	brew install shellcheck
-endif
-ifneq ($(OS),Linux)
-	uv tool install --upgrade codespell
-	uv tool install --upgrade ruff
-endif
-	uv tool install --upgrade yamllint
-	uv tool update-shell
+setup: install-uv setup-precommit ## Set up a development environment
+	uv sync --frozen --all-extras
+
+.PHONY: setup-tests
+setup-tests: install-uv install-venv  ##- Set up a testing environment without linters
+	uv sync --frozen
+
+.PHONY: setup-lint
+setup-lint: install-uv install-shellcheck install-ruff install-codespell ##- Set up a linting environment
+	uv sync --frozen --no-dev --no-install-workspace --extra lint --extra types
+
+.PHONY: setup-docs
+setup-docs: install-uv  ##- Set up a documentation-only environment
+	uv sync --frozen --no-dev --no-install-workspace --extra docs
 
 .PHONY: setup-precommit
-setup-precommit:  ## Set up pre-commit hooks in this repository.
-	uvx pre-commit install
+setup-precommit: install-uv  ##- Set up pre-commit hooks in this repository.
+ifeq ($(shell which pre-commit),)
+	uv tool install pre-commit
+endif
+	pre-commit install
+
+---------------- : ## ----------------
 
 .PHONY: format
 format: format-ruff format-codespell  ## Run all automatic formatters
 
-.PHONY: lint
-lint: lint-ruff lint-codespell lint-mypy lint-pyright lint-yaml  ## Run all linters
+.PHONY: format-ruff
+format-ruff:  ##- Automatically format with ruff
+	success=true
+	ruff check --fix $(SOURCES) || success=false
+	ruff format $(SOURCES)
+	$$success || exit 1
 
-.PHONY: test
-test: test-unit test-integration  ## Run all tests with the default python
+.PHONY: format-codespell
+format-codespell:  ##- Fix spelling issues with codespell
+	uv run codespell --toml pyproject.toml --write-changes $(SOURCES)
```

Comment:
Will it use the default `codespell` (installed as `snap` package above) ?

Reply:
Good catch!
```suggestion
	codespell --toml pyproject.toml --write-changes $(SOURCES)
```